### PR TITLE
MSPSDS-897: Reduce number of queries

### PR DESCRIPTION
**Pull request by kicferk1**
This solves the n+1 problems on case list page, case page and excel export sheet. 

Loading attachments on activities proved tricky, since not all of them have attachments, and not all attachments are on the same key, and so on. The current solution uses Preloader to only attempt to load attachments for activities that have them.

### CODE REVIEW

File: mspsds-web/app/controllers/investigations_controller.rb [hunk 1]
```diff
@@ -155,4 +173,18 @@ def respond_to_invalid_data(origin)
       format.json { render json: @investigation.errors, status: :unprocessable_entity }
     end
   end
+
+  def preload_activities
+    @activities = @investigation.activities.eager_load(:source)
+    preload_manually(@activities.select { |a| a.respond_to?("attachment") },
```
Comment (broder) [inline mspsds-web/app/controllers/investigations_controller.rb]: Should we do this for correspondence too?

File: mspsds-web/app/controllers/investigations_controller.rb [hunk 2]
```diff
@@ -111,6 +120,15 @@ def create
 
 private
 
+  def eager_load_investigation_data
```
Comment (broder) [inline mspsds-web/app/controllers/investigations_controller.rb]: Should we call this `set_investigation_with_associations` or something like that?

File: mspsds-web/app/controllers/investigations_controller.rb [hunk 3]
```diff
@@ -155,4 +173,20 @@ def respond_to_invalid_data(origin)
...
+                     [{ attachment_attachment: :blob }])
+    preload_manually(@activities.select { |a| a.respond_to?("email_file") },
+                     [{ email_file_attachment: :blob }, { email_attachment_attachment: :blob }])
+    preload_manually(@activities.select { |a| a.respond_to?("transcript") },
+                     [{ transcript_attachment: :blob }, { related_attachment_attachment: :blob }])
+    preload_manually(@activities.select { |a| a.respond_to?("correspondence") },
+                     [:correspondence])
+  end
+
+  def preload_manually(records, associations)
```
Comment (broder) [inline mspsds-web/app/controllers/investigations_controller.rb]: Is this the best place for this method? Would we want to use it for any other models?

(Sorry, I think my internet died when adding the comment the first time so it wasn't included in the review...)

Comment (kicferk1) [reply mspsds-web/app/controllers/investigations_controller.rb]: Not sure if we ever will want to use it elsewhere, but it does seem like it doesn't fit into investigation_controller.

**Pull request merged**